### ISSUE TITLE
DragAndDrop Calendar doesn't update resourceHeader properly

### ISSUE THREAD

**Issue by Bielik20** [jquense/react-big-calendar#1731]
#### Do you want to request a _feature_ or report a _bug_?

_bug_

#### What's the current behavior?

`resourceHeader` doesn't update properly in `DragAndDrop` Calendar. I have extracted simplified snipped from my code:

```tsx
function useResourceToggle<TResource>(
  resources: TResource[],
): [TResource[], (resource: TResource) => void] {
  const [selected, setSelected] = useState<TResource | null>(null);
  const selector = (resource: TResource) => {
    setSelected(selected ? null : resource);
  };

  return [selected ? [selected] : resources, selector];
}

export const Component = ({ assignments, users }: Props) => {
  const [resources, resourceToggle] = useResourceToggle(users);

  return (
    <BigCalendarDnD<Assignment, User>
      events={assignments}
      resources={resources}
      components={{
        resourceHeader: (props: any) => {
          return (
            <div onClick={() => resourceToggle(props.resource)}>{props.resource.displayName}</div>
          );
        },
      }}
    />
  );
};
```

`resourceToggle` function inside `resourceHeader` doesn't update. This results in inability to reset `selected`. In other words `selected` will always by `null` when `resourceToggle` is called from within `resourceHeader`.

This happens only with `DragAndDrop` plugin. Without it everything works fine.

#### What's the expected behavior?

`resourceHeader` should be updated properly.

**Comment by brianmwadime** [comment]
>#### Do you want to request a _feature_ or report a _bug_? ...

@Bielik20 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by cutterbl** [comment]
Do you have a CodeSandbox with what you're trying to do here? I'm trying to visualize it, but just not seeing it.

**Comment by Bielik20** [comment]
I don't have CodeSandbox at hand but I can show it on pictures.

### All resources:
![image](https://user-images.githubusercontent.com/13436218/94009734-288d4700-fda5-11ea-9bef-f2c75245cd89.png)

### Only one toggled resource:
![image](https://user-images.githubusercontent.com/13436218/94009799-3f339e00-fda5-11ea-8a8d-8caac6e1ada1.png)

**Comment by cutterbl** [comment]
So, when selecting a resource (header) you want it to only show that resource's items, right? Clicking again should show the other resources. Does that sound right? Just trying to fully understand your use case.

**Comment by Bielik20** [comment]
Yes, that is exactly the use case. I did it with very dirty workaround and I hope to remove it. It is clearly something with `DragAndDrop`, because it works fine without it.

**Comment by cutterbl** [comment]
I've written a few different component overrides, but never messed with the resource header before. As far as I can tell though, reading through the code I'm not seeing where the hoc does anything specific with that component. 

I also use the DnD hoc, and will be doing some stuff with resources in a future iteration of my work. Can't give you any clues now, but if I find something I'll post back (but, unfortunately, it won't be any time soon)

**Comment by Bielik20** [comment]
Thank you, in the mean time I will post my workaround for anyone who stumbles opon this issue:
```ts
export function useResourceToggle<TResource>(
  resources: TResource[],
): [TResource[], (resource: TResource) => void] {
  const [selected, setSelected] = useState<TResource | null>(null);
  const selector = (resource: TResource) => {
    setSelected(selected ? null : resource);
  };

  return [
    (selected ? [selected] : resources).map((res) => {
      // @ts-ignore
      res.__toggle = () => selector(res);
      return res;
    }),
    selector,
  ];
}
```

It is dirty and ugly but it works. You must use `__toggle` method on a resource to toggle, not the returned `selector` callback.
